mocked RestClient in startup bean - missing uri

### Describe the bug

In #47383, a problem was fixed where the uri of a restclient was required to be set - event though that rest client was a mock.
The fix works generally speaking. However, I discovered an edge case in my code base.

If the mocked rest client is called from an `@Startup` annotated normal scoped bean, I still get the same exception. Any my test fails.

### Expected behavior

Mocking Rest client should also work for startup beans, i.e. the uri is not required

### Actual behavior

```
C:\eclipse\tools\java\25-temurin\bin\java.exe -javaagent:C:\Users\mnp\AppData\Local\JetBrains\IntelliJIdea2025.3\captureAgent\debugger-agent.jar=file:///C:/Users/mnp/AppData/Local/Temp/capture8850392625203941756.props -ea -Djava.util.logging.manager=org.jboss.logmanager.LogManager -Dmaven.home=C:\Users\mnp\.m2\wrapper\dists\apache-maven-3.9.12-bin\5nmfsn99br87k5d4ajlekdq10k\apache-maven-3.9.12 --add-opens java.base/java.lang=ALL-UNNAMED -Dnative.image.path=C:\workspaces\projects\qbug-47383-ext\target/qbug-47383-ext-1.0.0-SNAPSHOT-runner --add-opens java.base/java.lang=ALL-UNNAMED -Didea.test.cyclic.buffer.size=1048576 "-javaagent:C:\Users\mnp\AppData\Local\Programs\IntelliJ IDEA Ultimate 2\lib\idea_rt.jar=59377" -Dkotlinx.coroutines.debug.enable.creation.stack.trace=false -Ddebugger.agent.enable.coroutines=true -Dkotlinx.coroutines.debug.enable.flows.stack.trace=true -Dkotlinx.coroutines.debug.enable.mutable.state.flows.stack.trace=true -Ddebugger.async.stack.trace.for.all.threads=true -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -classpath "C:\Users\mnp\AppData\Local\Programs\IntelliJ IDEA Ultimate 2\lib\idea_rt.jar;C:\Users\mnp\AppData\Local\Programs\IntelliJ IDEA Ultimate 2\plugins\junit\lib\junit6-rt.jar;C:\Users\mnp\AppData\Local\Programs\IntelliJ IDEA Ultimate 2\plugins\junit\lib\junit5-rt.jar;C:\Users\mnp\AppData\Local\Programs\IntelliJ IDEA Ultimate 2\plugins\junit\lib\junit-rt.jar;C:\workspaces\projects\qbug-47383-ext\target\test-classes;C:\workspaces\projects\qbug-47383-ext\target\classes;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-client-jackson\3.30.5\quarkus-rest-client-jackson-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive-jackson\3.30.5\resteasy-reactive-jackson-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive\3.30.5\resteasy-reactive-3.30.5.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\core\jackson-databind\2.20.1\jackson-databind-2.20.1.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.20\jackson-annotations-2.20.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\core\jackson-core\2.20.1\jackson-core-2.20.1.jar;C:\Users\mnp\.m2\repository\jakarta\annotation\jakarta.annotation-api\3.0.0\jakarta.annotation-api-3.0.0.jar;C:\Users\mnp\.m2\repository\org\jboss\logging\commons-logging-jboss-logging\1.0.0.Final\commons-logging-jboss-logging-1.0.0.Final.jar;C:\Users\mnp\.m2\repository\jakarta\xml\bind\jakarta.xml.bind-api\4.0.4\jakarta.xml.bind-api-4.0.4.jar;C:\Users\mnp\.m2\repository\jakarta\activation\jakarta.activation-api\2.1.4\jakarta.activation-api-2.1.4.jar;C:\Users\mnp\.m2\repository\org\jboss\logging\jboss-logging\3.6.1.Final\jboss-logging-3.6.1.Final.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-jackson-common\3.30.5\quarkus-rest-jackson-common-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-jackson\3.30.5\quarkus-jackson-3.30.5.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jsr310\2.20.1\jackson-datatype-jsr310-2.20.1.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jdk8\2.20.1\jackson-datatype-jdk8-2.20.1.jar;C:\Users\mnp\.m2\repository\com\fasterxml\jackson\module\jackson-module-parameter-names\2.20.1\jackson-module-parameter-names-2.20.1.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-client\3.30.5\quarkus-rest-client-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-client-jaxrs\3.30.5\quarkus-rest-client-jaxrs-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-vertx\3.30.5\quarkus-vertx-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-netty\3.30.5\quarkus-netty-3.30.5.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec\4.1.130.Final\netty-codec-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec-haproxy\4.1.130.Final\netty-codec-haproxy-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-annotation\2.14.0\smallrye-common-annotation-2.14.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-vertx-latebound-mdc-provider\3.30.5\quarkus-vertx-latebound-mdc-provider-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\smallrye-fault-tolerance-vertx\6.9.3\smallrye-fault-tolerance-vertx-6.9.3.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive-client\3.30.5\resteasy-reactive-client-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\stork\stork-api\2.7.3\stork-api-2.7.3.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-web-client\4.5.23\vertx-web-client-4.5.23.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-uri-template\4.5.23\vertx-uri-template-4.5.23.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-client-config\3.30.5\quarkus-rest-client-config-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\jandex\3.5.2\jandex-3.5.2.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-proxy-registry\3.30.5\quarkus-proxy-registry-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-credentials\3.30.5\quarkus-credentials-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-tls-registry\3.30.5\quarkus-tls-registry-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-tls-registry-spi\3.30.5\quarkus-tls-registry-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\certs\smallrye-private-key-pem-parser\0.9.2\smallrye-private-key-pem-parser-0.9.2.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-constraint\2.14.0\smallrye-common-constraint-2.14.0.jar;C:\Users\mnp\.m2\repository\org\eclipse\microprofile\rest\client\microprofile-rest-client-api\4.0\microprofile-rest-client-api-4.0.jar;C:\Users\mnp\.m2\repository\jakarta\inject\jakarta.inject-api\2.0.1\jakarta.inject-api-2.0.1.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-arc\3.30.5\quarkus-arc-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\arc\arc\3.30.5\arc-3.30.5.jar;C:\Users\mnp\.m2\repository\jakarta\enterprise\jakarta.enterprise.cdi-api\4.1.0\jakarta.enterprise.cdi-api-4.1.0.jar;C:\Users\mnp\.m2\repository\jakarta\enterprise\jakarta.enterprise.lang-model\4.1.0\jakarta.enterprise.lang-model-4.1.0.jar;C:\Users\mnp\.m2\repository\jakarta\el\jakarta.el-api\6.0.1\jakarta.el-api-6.0.1.jar;C:\Users\mnp\.m2\repository\jakarta\interceptor\jakarta.interceptor-api\2.2.0\jakarta.interceptor-api-2.2.0.jar;C:\Users\mnp\.m2\repository\jakarta\transaction\jakarta.transaction-api\2.0.1\jakarta.transaction-api-2.0.1.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\mutiny\3.1.0\mutiny-3.1.0.jar;C:\Users\mnp\.m2\repository\org\jctools\jctools-core\4.0.5\jctools-core-4.0.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-core\3.30.5\quarkus-core-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-os\2.14.0\smallrye-common-os-2.14.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-ide-launcher\3.30.5\quarkus-ide-launcher-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-development-mode-spi\3.30.5\quarkus-development-mode-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\config\smallrye-config\3.14.1\smallrye-config-3.14.1.jar;C:\Users\mnp\.m2\repository\io\smallrye\config\smallrye-config-core\3.14.1\smallrye-config-core-3.14.1.jar;C:\Users\mnp\.m2\repository\org\eclipse\microprofile\config\microprofile-config-api\3.1\microprofile-config-api-3.1.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-classloader\2.14.0\smallrye-common-classloader-2.14.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\config\smallrye-config-common\3.14.1\smallrye-config-common-3.14.1.jar;C:\Users\mnp\.m2\repository\org\jboss\logmanager\jboss-logmanager\3.1.2.Final\jboss-logmanager-3.1.2.Final.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-cpu\2.14.0\smallrye-common-cpu-2.14.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-expression\2.14.0\smallrye-common-expression-2.14.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-net\2.14.0\smallrye-common-net-2.14.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-ref\2.14.0\smallrye-common-ref-2.14.0.jar;C:\Users\mnp\.m2\repository\jakarta\json\jakarta.json-api\2.1.3\jakarta.json-api-2.1.3.jar;C:\Users\mnp\.m2\repository\org\jboss\threads\jboss-threads\3.9.2\jboss-threads-3.9.2.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-function\2.14.0\smallrye-common-function-2.14.0.jar;C:\Users\mnp\.m2\repository\org\slf4j\slf4j-api\2.0.17\slf4j-api-2.0.17.jar;C:\Users\mnp\.m2\repository\org\jboss\slf4j\slf4j-jboss-logmanager\2.0.2.Final\slf4j-jboss-logmanager-2.0.2.Final.jar;C:\Users\mnp\.m2\repository\org\wildfly\common\wildfly-common\2.0.1\wildfly-common-2.0.1.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-runner\3.30.5\quarkus-bootstrap-runner-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-fs-util\1.2.0\quarkus-fs-util-1.2.0.jar;C:\Users\mnp\.m2\repository\org\eclipse\microprofile\context-propagation\microprofile-context-propagation-api\1.3\microprofile-context-propagation-api-1.3.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest\3.30.5\quarkus-rest-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-rest-common\3.30.5\quarkus-rest-common-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive-common\3.30.5\resteasy-reactive-common-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive-common-types\3.30.5\resteasy-reactive-common-types-3.30.5.jar;C:\Users\mnp\.m2\repository\org\reactivestreams\reactive-streams\1.0.4\reactive-streams-1.0.4.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\mutiny-zero-flow-adapters\1.1.1\mutiny-zero-flow-adapters-1.1.1.jar;C:\Users\mnp\.m2\repository\io\quarkus\resteasy\reactive\resteasy-reactive-vertx\3.30.5\resteasy-reactive-vertx-3.30.5.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-web\4.5.23\vertx-web-4.5.23.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-web-common\4.5.23\vertx-web-common-4.5.23.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-auth-common\4.5.23\vertx-auth-common-4.5.23.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-bridge-common\4.5.23\vertx-bridge-common-4.5.23.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-core\3.21.3\smallrye-mutiny-vertx-core-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-runtime\3.21.3\smallrye-mutiny-vertx-runtime-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\vertx-mutiny-generator\3.21.3\vertx-mutiny-generator-3.21.3.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-codegen\4.5.23\vertx-codegen-4.5.23.jar;C:\Users\mnp\.m2\repository\io\quarkus\vertx\utils\quarkus-vertx-utils\3.30.5\quarkus-vertx-utils-3.30.5.jar;C:\Users\mnp\.m2\repository\jakarta\ws\rs\jakarta.ws.rs-api\3.1.0\jakarta.ws.rs-api-3.1.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-vertx-http\3.30.5\quarkus-vertx-http-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-security-runtime-spi\3.30.5\quarkus-security-runtime-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-vertx-context\2.14.0\smallrye-common-vertx-context-2.14.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\security\quarkus-security\2.2.1\quarkus-security-2.2.1.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-web\3.21.3\smallrye-mutiny-vertx-web-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-web-common\3.21.3\smallrye-mutiny-vertx-web-common-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-auth-common\3.21.3\smallrye-mutiny-vertx-auth-common-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-bridge-common\3.21.3\smallrye-mutiny-vertx-bridge-common-3.21.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\smallrye-mutiny-vertx-uri-template\3.21.3\smallrye-mutiny-vertx-uri-template-3.21.3.jar;C:\Users\mnp\.m2\repository\org\crac\crac\1.5.0\crac-1.5.0.jar;C:\Users\mnp\.m2\repository\com\aayushatharva\brotli4j\brotli4j\1.16.0\brotli4j-1.16.0.jar;C:\Users\mnp\.m2\repository\com\aayushatharva\brotli4j\service\1.16.0\service-1.16.0.jar;C:\Users\mnp\.m2\repository\com\aayushatharva\brotli4j\native-windows-x86_64\1.16.0\native-windows-x86_64-1.16.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-jsonp\3.30.5\quarkus-jsonp-3.30.5.jar;C:\Users\mnp\.m2\repository\org\eclipse\parsson\parsson\1.1.7\parsson-1.1.7.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-virtual-threads\3.30.5\quarkus-virtual-threads-3.30.5.jar;C:\Users\mnp\.m2\repository\io\vertx\vertx-core\4.5.23\vertx-core-4.5.23.jar;C:\Users\mnp\.m2\repository\io\netty\netty-common\4.1.130.Final\netty-common-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-buffer\4.1.130.Final\netty-buffer-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-transport\4.1.130.Final\netty-transport-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-handler\4.1.130.Final\netty-handler-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-transport-native-unix-common\4.1.130.Final\netty-transport-native-unix-common-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-handler-proxy\4.1.130.Final\netty-handler-proxy-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec-socks\4.1.130.Final\netty-codec-socks-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec-http\4.1.130.Final\netty-codec-http-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec-http2\4.1.130.Final\netty-codec-http2-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-resolver\4.1.130.Final\netty-resolver-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-resolver-dns\4.1.130.Final\netty-resolver-dns-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\netty\netty-codec-dns\4.1.130.Final\netty-codec-dns-4.1.130.Final.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-junit5\3.30.5\quarkus-junit5-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-core\3.30.5\quarkus-bootstrap-core-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-classloader-commons\3.30.5\quarkus-classloader-commons-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-app-model\3.30.5\quarkus-bootstrap-app-model-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-io\2.14.0\smallrye-common-io-2.14.0.jar;C:\Users\mnp\.m2\repository\org\eclipse\sisu\org.eclipse.sisu.inject\0.9.0.M4\org.eclipse.sisu.inject-0.9.0.M4.jar;C:\Users\mnp\.m2\repository\org\ow2\asm\asm\9.9\asm-9.9.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-test-common\3.30.5\quarkus-test-common-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-core-deployment\3.30.5\quarkus-core-deployment-3.30.5.jar;C:\Users\mnp\.m2\repository\org\aesh\readline\2.6\readline-2.6.jar;C:\Users\mnp\.m2\repository\org\fusesource\jansi\jansi\2.4.0\jansi-2.4.0.jar;C:\Users\mnp\.m2\repository\org\aesh\aesh\2.8.2\aesh-2.8.2.jar;C:\Users\mnp\.m2\repository\org\apache\commons\commons-compress\1.28.0\commons-compress-1.28.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\gizmo\gizmo\1.9.0\gizmo-1.9.0.jar;C:\Users\mnp\.m2\repository\org\ow2\asm\asm-util\9.9\asm-util-9.9.jar;C:\Users\mnp\.m2\repository\org\ow2\asm\asm-analysis\9.9\asm-analysis-9.9.jar;C:\Users\mnp\.m2\repository\io\quarkus\gizmo\gizmo2\2.0.0.Beta10\gizmo2-2.0.0.Beta10.jar;C:\Users\mnp\.m2\repository\io\github\dmlloyd\jdk-classfile-backport\25.1\jdk-classfile-backport-25.1.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-resource\2.14.0\smallrye-common-resource-2.14.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\jandex-gizmo2\3.5.2\jandex-gizmo2-3.5.2.jar;C:\Users\mnp\.m2\repository\io\smallrye\common\smallrye-common-process\2.14.0\smallrye-common-process-2.14.0.jar;C:\Users\mnp\.m2\repository\org\ow2\asm\asm-commons\9.9\asm-commons-9.9.jar;C:\Users\mnp\.m2\repository\org\ow2\asm\asm-tree\9.9\asm-tree-9.9.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-hibernate-validator-spi\3.30.5\quarkus-hibernate-validator-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-class-change-agent\3.30.5\quarkus-class-change-agent-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-builder\3.30.5\quarkus-builder-3.30.5.jar;C:\Users\mnp\.m2\repository\org\graalvm\sdk\nativeimage\23.1.2\nativeimage-23.1.2.jar;C:\Users\mnp\.m2\repository\org\graalvm\sdk\word\23.1.2\word-23.1.2.jar;C:\Users\mnp\.m2\repository\org\junit\platform\junit-platform-launcher\1.13.4\junit-platform-launcher-1.13.4.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-maven-resolver\3.30.5\quarkus-bootstrap-maven-resolver-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\beanbag\smallrye-beanbag-maven\1.5.3\smallrye-beanbag-maven-1.5.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\beanbag\smallrye-beanbag-sisu\1.5.3\smallrye-beanbag-sisu-1.5.3.jar;C:\Users\mnp\.m2\repository\io\smallrye\beanbag\smallrye-beanbag\1.5.3\smallrye-beanbag-1.5.3.jar;C:\Users\mnp\.m2\repository\javax\inject\javax.inject\1\javax.inject-1.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-artifact\3.9.12\maven-artifact-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-builder-support\3.9.12\maven-builder-support-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-model\3.9.12\maven-model-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-model-builder\3.9.12\maven-model-builder-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-repository-metadata\3.9.12\maven-repository-metadata-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-settings\3.9.12\maven-settings-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-api\1.9.25\maven-resolver-api-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-impl\1.9.25\maven-resolver-impl-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-named-locks\1.9.25\maven-resolver-named-locks-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-spi\1.9.25\maven-resolver-spi-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-util\1.9.25\maven-resolver-util-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-transport-http\1.9.23\maven-resolver-transport-http-1.9.23.jar;C:\Users\mnp\.m2\repository\org\apache\maven\wagon\wagon-provider-api\3.5.3\wagon-provider-api-3.5.3.jar;C:\Users\mnp\.m2\repository\org\apache\maven\wagon\wagon-http-shared\3.5.3\wagon-http-shared-3.5.3.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-interpolation\1.26\plexus-interpolation-1.26.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-utils\3.5.1\plexus-utils-3.5.1.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-xml\4.0.2\plexus-xml-4.0.2.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-xml-impl\4.0.0-alpha-7\maven-xml-impl-4.0.0-alpha-7.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-api-xml\4.0.0-alpha-7\maven-api-xml-4.0.0-alpha-7.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-api-meta\4.0.0-alpha-7\maven-api-meta-4.0.0-alpha-7.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-cipher\2.0\plexus-cipher-2.0.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-sec-dispatcher\2.0\plexus-sec-dispatcher-2.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-maven4-resolver\3.30.5\quarkus-bootstrap-maven4-resolver-3.30.5.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-embedder\3.9.12\maven-embedder-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-core\3.9.12\maven-core-3.9.12.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-component-annotations\2.1.0\plexus-component-annotations-2.1.0.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-plugin-api\3.9.12\maven-plugin-api-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\shared\maven-shared-utils\3.4.2\maven-shared-utils-3.4.2.jar;C:\Users\mnp\.m2\repository\com\google\inject\guice\5.1.0\guice-5.1.0-classes.jar;C:\Users\mnp\.m2\repository\aopalliance\aopalliance\1.0\aopalliance-1.0.jar;C:\Users\mnp\.m2\repository\com\google\guava\guava\33.5.0-jre\guava-33.5.0-jre.jar;C:\Users\mnp\.m2\repository\org\jspecify\jspecify\1.0.0\jspecify-1.0.0.jar;C:\Users\mnp\.m2\repository\com\google\guava\failureaccess\1.0.1\failureaccess-1.0.1.jar;C:\Users\mnp\.m2\repository\javax\annotation\javax.annotation-api\1.3.2\javax.annotation-api-1.3.2.jar;C:\Users\mnp\.m2\repository\org\codehaus\plexus\plexus-classworlds\2.6.0\plexus-classworlds-2.6.0.jar;C:\Users\mnp\.m2\repository\commons-cli\commons-cli\1.11.0\commons-cli-1.11.0.jar;C:\Users\mnp\.m2\repository\org\eclipse\sisu\org.eclipse.sisu.plexus\0.9.0.M4\org.eclipse.sisu.plexus-0.9.0.M4.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-settings-builder\3.9.12\maven-settings-builder-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\maven-resolver-provider\3.9.12\maven-resolver-provider-3.9.12.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-connector-basic\1.9.25\maven-resolver-connector-basic-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\resolver\maven-resolver-transport-wagon\1.9.25\maven-resolver-transport-wagon-1.9.25.jar;C:\Users\mnp\.m2\repository\org\apache\maven\wagon\wagon-http\3.5.3\wagon-http-3.5.3.jar;C:\Users\mnp\.m2\repository\org\apache\maven\wagon\wagon-file\3.5.3\wagon-file-3.5.3.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-bootstrap-gradle-resolver\3.30.5\quarkus-bootstrap-gradle-resolver-3.30.5.jar;C:\Users\mnp\.m2\repository\commons-io\commons-io\2.21.0\commons-io-2.21.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-junit5-config\3.30.5\quarkus-junit5-config-3.30.5.jar;C:\Users\mnp\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.13.4\junit-jupiter-api-5.13.4.jar;C:\Users\mnp\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\mnp\.m2\repository\org\junit\platform\junit-platform-commons\1.13.4\junit-platform-commons-1.13.4.jar;C:\Users\mnp\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\mnp\.m2\repository\org\junit\jupiter\junit-jupiter\5.13.4\junit-jupiter-5.13.4.jar;C:\Users\mnp\.m2\repository\org\junit\jupiter\junit-jupiter-params\5.13.4\junit-jupiter-params-5.13.4.jar;C:\Users\mnp\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.13.4\junit-jupiter-engine-5.13.4.jar;C:\Users\mnp\.m2\repository\org\junit\platform\junit-platform-engine\1.13.4\junit-platform-engine-1.13.4.jar;C:\Users\mnp\.m2\repository\io\rest-assured\rest-assured\5.5.6\rest-assured-5.5.6.jar;C:\Users\mnp\.m2\repository\org\apache\groovy\groovy\4.0.22\groovy-4.0.22.jar;C:\Users\mnp\.m2\repository\org\apache\groovy\groovy-xml\4.0.22\groovy-xml-4.0.22.jar;C:\Users\mnp\.m2\repository\org\apache\httpcomponents\httpclient\4.5.14\httpclient-4.5.14.jar;C:\Users\mnp\.m2\repository\org\apache\httpcomponents\httpcore\4.4.16\httpcore-4.4.16.jar;C:\Users\mnp\.m2\repository\commons-codec\commons-codec\1.20.0\commons-codec-1.20.0.jar;C:\Users\mnp\.m2\repository\org\apache\httpcomponents\httpmime\4.5.14\httpmime-4.5.14.jar;C:\Users\mnp\.m2\repository\org\hamcrest\hamcrest\2.2\hamcrest-2.2.jar;C:\Users\mnp\.m2\repository\org\ccil\cowan\tagsoup\tagsoup\1.2.1\tagsoup-1.2.1.jar;C:\Users\mnp\.m2\repository\io\rest-assured\json-path\5.5.6\json-path-5.5.6.jar;C:\Users\mnp\.m2\repository\org\apache\groovy\groovy-json\4.0.22\groovy-json-4.0.22.jar;C:\Users\mnp\.m2\repository\io\rest-assured\rest-assured-common\5.5.6\rest-assured-common-5.5.6.jar;C:\Users\mnp\.m2\repository\io\rest-assured\xml-path\5.5.6\xml-path-5.5.6.jar;C:\Users\mnp\.m2\repository\org\apache\commons\commons-lang3\3.19.0\commons-lang3-3.19.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-junit5-mockito\3.30.5\quarkus-junit5-mockito-3.30.5.jar;C:\Users\mnp\.m2\repository\org\mockito\mockito-junit-jupiter\5.20.0\mockito-junit-jupiter-5.20.0.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-junit5-mockito-config\3.30.5\quarkus-junit5-mockito-config-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-arc-deployment\3.30.5\quarkus-arc-deployment-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-smallrye-context-propagation-spi\3.30.5\quarkus-smallrye-context-propagation-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-devui-deployment-spi\3.30.5\quarkus-devui-deployment-spi-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-arc-dev\3.30.5\quarkus-arc-dev-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\arc\arc-processor\3.30.5\arc-processor-3.30.5.jar;C:\Users\mnp\.m2\repository\org\mockito\mockito-core\5.20.0\mockito-core-5.20.0.jar;C:\Users\mnp\.m2\repository\net\bytebuddy\byte-buddy\1.17.6\byte-buddy-1.17.6.jar;C:\Users\mnp\.m2\repository\net\bytebuddy\byte-buddy-agent\1.17.7\byte-buddy-agent-1.17.7.jar;C:\Users\mnp\.m2\repository\org\objenesis\objenesis\3.3\objenesis-3.3.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-mutiny\3.30.5\quarkus-mutiny-3.30.5.jar;C:\Users\mnp\.m2\repository\io\quarkus\quarkus-smallrye-context-propagation\3.30.5\quarkus-smallrye-context-propagation-3.30.5.jar;C:\Users\mnp\.m2\repository\io\smallrye\smallrye-context-propagation\2.3.0\smallrye-context-propagation-2.3.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\smallrye-context-propagation-api\2.3.0\smallrye-context-propagation-api-2.3.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\smallrye-context-propagation-storage\2.3.0\smallrye-context-propagation-storage-2.3.0.jar;C:\Users\mnp\.m2\repository\io\smallrye\reactive\mutiny-smallrye-context-propagation\3.1.0\mutiny-smallrye-context-propagation-3.1.0.jar" com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit5 org.acme.GreetingServiceTest

java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus

	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:663)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:758)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:350)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:229)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:637)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:682)
	... 1 more
Caused by: java.lang.IllegalArgumentException: Unable to determine the proper baseUrl/baseUri. Consider registering using @RegisterRestClient(baseUri="someuri"), @RegisterRestClient(configKey="orkey"), or by adding 'quarkus.rest-client."org.acme.MyRemoteService".url' or 'quarkus.rest-client."org.acme.MyRemoteService".uri' to your Quarkus configuration
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.configureBaseUrl(RestClientCDIDelegateBuilder.java:434)
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.configureBuilder(RestClientCDIDelegateBuilder.java:80)
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.build(RestClientCDIDelegateBuilder.java:75)
	at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.createDelegate(RestClientCDIDelegateBuilder.java:56)
	at io.quarkus.rest.client.reactive.runtime.RestClientReactiveCDIWrapperBase.constructDelegate(RestClientReactiveCDIWrapperBase.java:75)
	at io.quarkus.rest.client.reactive.runtime.RestClientReactiveCDIWrapperBase.getDelegate(RestClientReactiveCDIWrapperBase.java:57)
	at org.acme.MyRemoteService$$CDIWrapper.getGreeting(Unknown Source)
	at org.acme.MyRemoteService$$CDIWrapper_ClientProxy.getGreeting(Unknown Source)
	at org.acme.ServiceB.getGreeting(ServiceB.java:13)
	at org.acme.ServiceB_ClientProxy.getGreeting(Unknown Source)
	at org.acme.ServiceA.onStart(ServiceA.java:19)
	at org.acme.ServiceA_Bean.create(Unknown Source)
	at org.acme.ServiceA_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.ch4(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:23)
	at org.acme.ServiceA_ClientProxy.arc$delegate(Unknown Source)
	at org.acme.ServiceA_ClientProxy.arc_contextualInstance(Unknown Source)
	at org.acme.ServiceA_Observer_Synthetic_6OggtSI-m7L2fZTpi_UdmBb5AA8.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:366)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:348)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:81)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:165)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:115)
	at io.quarkus.runner.recorded.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.runner.recorded.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 8 more


Process finished with exit code -1
```

### How to Reproduce?

https://github.com/Postremus/qbug-47383-ext.git 

Execute GreetingServiceTest, which will fail

The test will succeed if you remove the `@Startup` from ServiceA.

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-26100 NANBCHL9NG3 3.3.6-341.x86_64 2022-09-05 20:28 UTC x86_64 Msys

### Output of `java -version`

openjdk 25.0.1 2025-10-21 LTS OpenJDK Runtime Environment Temurin-25.0.1+8 (build 25.0.1+8-LTS) OpenJDK 64-Bit Server VM Temurin-25.0.1+8 (build 25.0.1+8-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.30.5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.12 (848fbb4bf2d427b72bdb2471c22fced7ebd9a7a1) Maven home: C:\eclipse\tools\java\maven Java version: 25.0.1, vendor: Eclipse Adoptium, runtime: C:\eclipse\tools\java\25-temurin Default locale: de_DE, platform encoding: UTF-8 OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/docs/src/main/asciidoc/getting-started-testing.adoc
+++ b/docs/src/main/asciidoc/getting-started-testing.adoc
@@ -712,6 +712,7 @@ However, note that it is not possible to assign different test profiles or resou
 Quarkus supports the use of mock objects using two different approaches. You can either use CDI alternatives to
 mock out a bean for all test classes, or use `QuarkusMock` to mock out beans on a per test basis.
 
+[[mock-cdi-alternative]]
 === CDI `@Alternative` mechanism.
 
 To use this simply override the bean you wish to mock with a class in the `src/test/java` directory, and put the `@Alternative` and `@Priority(1)` annotations on the bean.
@@ -1066,6 +1067,13 @@ public class GreetingResourceTest {
 }
 ----
 <1> Indicate that this injection point is meant to use an instance of `RestClient`.
+
+=== Mocking on Startup
+
+Beans accessed and used during startup are not subject to automatic mocking, even if the test declares injection points
+with the same bean types annotated with @InjectMock or @InjectSpy. Mocking is a local operation within a test, whereas
+startup is a global operation for executing many tests. In such cases, the recommendation is to use the
+<<mock-cdi-alternative>>.
 
 === Mocking with Panache
 

--- a/integration-tests/injectmock/pom.xml
+++ b/integration-tests/injectmock/pom.xml
@@ -20,6 +20,10 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-jackson</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit-mockito</artifactId>
             <scope>test</scope>
         </dependency>
@@ -33,6 +37,19 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-rest-jackson-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-jackson-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/GreetingRestClient.java
+++ b/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/GreetingRestClient.java
@@ -1,0 +1,16 @@
+package io.quarkus.it.mockbean;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@RegisterRestClient
+public interface GreetingRestClient {
+    @GET
+    @Path("/hello")
+    @Produces(MediaType.TEXT_PLAIN)
+    String greeting();
+}

--- a/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/StartupService.java
+++ b/integration-tests/injectmock/src/main/java/io/quarkus/it/mockbean/StartupService.java
@@ -1,0 +1,32 @@
+package io.quarkus.it.mockbean;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import io.quarkus.runtime.Startup;
+
+@Startup
+@ApplicationScoped
+public class StartupService {
+    @Inject
+    @RestClient
+    GreetingRestClient greetingRestClient;
+
+    String greetingStartup = null;
+
+    @PostConstruct
+    void onStart() {
+        greetingStartup = greetingRestClient.greeting();
+    }
+
+    public String greetingStartup() {
+        return greetingStartup;
+    }
+
+    public String greeting() {
+        return greetingRestClient.greeting();
+    }
+}

--- a/integration-tests/injectmock/src/test/java/io/quarkus/it/mockbean/StartupTest.java
+++ b/integration-tests/injectmock/src/test/java/io/quarkus/it/mockbean/StartupTest.java
@@ -1,0 +1,43 @@
+package io.quarkus.it.mockbean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+class StartupTest {
+    @InjectMock
+    @RestClient
+    GreetingRestClient greetingRestClient;
+    @Inject
+    StartupService startupService;
+
+    @Test
+    void startup() {
+        Mockito.when(greetingRestClient.greeting()).thenReturn("Hello from Mocked REST Client");
+        assertEquals("Hello from Startup Mocked REST Client", startupService.greetingStartup());
+        assertEquals("Hello from Mocked REST Client", startupService.greeting());
+    }
+
+    @Produces
+    @ApplicationScoped
+    @RestClient
+    @Alternative
+    @Priority(1)
+    public GreetingRestClient restClient() {
+        GreetingRestClient mock = Mockito.mock(GreetingRestClient.class);
+        Mockito.when(mock.greeting()).thenReturn("Hello from Startup Mocked REST Client");
+        return mock;
+    }
+}

